src/Xabe.FFmpeg/Conversion/Conversion.cs

```diff
@@ -281,29 +284,79 @@ public IConversion SetOutput(string outputPath)
             OutputFilePath = outputPath;
             _output = $"\"{outputPath}\"";
             return this;
+        }
+        /// <inheritdoc />
+        public IConversion SetVSync(int vsyncmode)
+        {
+            if (vsyncmode != -1) AddParameter($"-vsync {vsyncmode}");
+            return this;
+        }
+        /// <inheritdoc />
+        public IConversion SetVideoBitrate(long bitrate)
+        {
+            AddParameter(string.Format("-b:v {0}", bitrate));
+            AddParameter(string.Format("-minrate {0}", bitrate));
+            AddParameter(string.Format("-maxrate {0}", bitrate));
+            AddParameter(string.Format("-bufsize {0}", bitrate));
+
+            if (HasH264Stream())
+                AddParameter("-x264opts nal-hrd=cbr:force-cfr=1");
+            // This may be possibly wrong as it assumes if the input stream is X264 and the output is software based x264. 
+            // What about if the output is hardware encoding X264 / X265.
+            // See Method of working out by supplying the output codec type and checking it directly.
+            return this;
+        }
+        /// <inheritdoc />
+        public IConversion SetVideoBitrate(string Minibitrate, string Maxbitrate, string Buffersize, VideoCodec Outputcodec)
+        {
+            AddParameter(string.Format("-b:v {0}", Minibitrate));
+            AddParameter(string.Format("-maxrate {0}", Maxbitrate));
+            AddParameter(string.Format("-bufsize {0}", Buffersize));
+
+            if (Outputcodec == VideoCodec.h264) AddParameter("-x264opts nal-hrd=cbr:force-cfr=1");
+
+            return this;
         }
-
-        /// <inheritdoc />
-        public IConversion SetVideoBitrate(long bitrate)
-        {
-            AddParameter(string.Format("-b:v {0}", bitrate));
-            AddParameter(string.Format("-minrate {0}", bitrate));
-            AddParameter(string.Format("-maxrate {0}", bitrate));
-            AddParameter(string.Format("-bufsize {0}", bitrate));
-
-            if (HasH264Stream())
-                AddParameter("-x264opts nal-hrd=cbr:force-cfr=1");
-
-            return this;
-        }
-
         /// <inheritdoc />
         public IConversion SetAudioBitrate(long bitrate)
         {
             AddParameter($"-b:a {bitrate} ");
             return this;
-        }
-
+        } 
+        /// <inheritdoc />
+        public IConversion SetMapChapters(string inputfile)
+        {
+            AddParameter(inputfile == "" ? " -map_chapters -1 " : " -map_chapters " + inputfile);
+            return this;
+        }
+        /// <inheritdoc />
+        public IConversion SetAudioCodecMode(AudioEncoder AE, string bitrate = "", object extraparams = null)

```

Comment:
Look at method BuildAudioCodec
In AudioStream.
I'm not sure if it shouldn't be there.

Reply:
Moved to build BuildAudioCodec, added extraparams as a extra param=null in SetAudioCodec.